query/sources/impl/src/main/java/org/codice/ddf/admin/sources/opensearch/persist/SaveOpenSearchConfiguration.java

```diff
@@ -0,0 +1,91 @@
+import static org.codice.ddf.admin.common.message.DefaultMessages.noExistingConfigError;
+import static org.codice.ddf.admin.common.services.ServiceCommons.createManagedService;
+import static org.codice.ddf.admin.common.services.ServiceCommons.serviceConfigurationExists;
+import static org.codice.ddf.admin.common.services.ServiceCommons.updateService;
+import static org.codice.ddf.admin.sources.commons.utils.SourceValidationUtils.validateSourceName;
+import static org.codice.ddf.admin.sources.services.OpenSearchServiceProperties.OPENSEARCH_FACTORY_PID;
+import static org.codice.ddf.admin.sources.services.OpenSearchServiceProperties.openSearchConfigToServiceProps;
+
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.admin.api.fields.Field;
+import org.codice.ddf.admin.common.actions.BaseAction;
+import org.codice.ddf.admin.common.fields.base.scalar.BooleanField;
+import org.codice.ddf.admin.common.fields.common.PidField;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.sources.fields.type.OpenSearchSourceConfigurationField;
+
+import com.google.common.collect.ImmutableList;
+
+public class SaveOpenSearchConfiguration extends BaseAction<BooleanField> {
+
+    public static final String ID = "saveOpenSearchSource";
+
+    public static final String DESCRIPTION =
+            "Saves an OpenSearch source configuration. If a pid is specified, the source configuration specified by the pid will be updated. Returns true on success and false on failure.";
+
+    private OpenSearchSourceConfigurationField config;
+
+    private PidField pid;
+
+    private ConfiguratorFactory configuratorFactory;
+
+
+    public SaveOpenSearchConfiguration(ConfiguratorFactory configuratorFactory) {
+        super(ID, DESCRIPTION, new BooleanField());
+        config = new OpenSearchSourceConfigurationField();
+        pid = new PidField();
+        config.isRequired(true);
+        config.sourceNameField().isRequired(true);
+        config.endpointUrlField().isRequired(true);
+        this.configuratorFactory = configuratorFactory;
+    }
+
+    @Override
+    public BooleanField performAction() {
+        if (StringUtils.isNotEmpty(pid.getValue())) {
+            addMessages(updateService(pid, openSearchConfigToServiceProps(config), configuratorFactory));
+        } else {
+            if(createManagedService(openSearchConfigToServiceProps(config), OPENSEARCH_FACTORY_PID, configuratorFactory).containsErrorMsgs()) {
+                addArgumentMessage(failedPersistError(config.path()));
+            }
+        }
+        return new BooleanField(!containsErrorMsgs());
+    }
+
+    @Override
+    public void validate() {
+        super.validate();
+        if(containsErrorMsgs()) {
+            return;
+        }
+
+        if(pid.getValue() != null && !serviceConfigurationExists(pid.getValue(), configuratorFactory)) {
```

Comment:
Consider changing serviceConfigurationExsits to return a report so everyone doesn't have to continually add the  `noExistingConfigError` msg.